packages/grafana-ui/src/components/Logs/LogRowMessage.tsx

```diff
@@ -63,17 +63,18 @@ function renderLogMessage(
 ) {
   const needsHighlighter =
     highlights && highlights.length > 0 && highlights[0] && highlights[0].length > 0 && entry.length < MAX_CHARACTERS;
-  if (needsHighlighter) {
+  const searchWords = highlights ?? [];
+  if (hasAnsi) {
+    return <LogMessageAnsi value={entry} highlight={{ searchWords, highlightClassName }} />;
```

Comment:
i think we should use the `needsHighlighter` boolean in the ansi-highlight case too, to be in sync with the other branches, something like this:

```suggestion
    const highlight = needsHighlighter ? {searchWords, highlightClassName} : undefined;
    return <LogMessageAnsi value={entry} highlight={highlight} />;
```